core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -58,6 +59,28 @@ final class JDBCSourceAsColumnarStore(_connProperties: ConnectionProperties,
     }
   }
 
+  override protected def doInsert(tableName: String, batch: CachedBatch,
+      batchId: UUID, partitionId: Int): (Connection => Any) = {
+    {
+      (connection: Connection) => {
+        super.doInsert(tableName, batch, batchId, partitionId)(connection)
+        connectionType match {
+          case ConnectionType.Embedded =>
+            val resolvedName = ExternalStoreUtils.lookupName(tableName,
+              connection.getSchema)
+            val region = Misc.getRegionForTable(resolvedName, true)
+            region.asInstanceOf[AbstractRegion] match {
```

Comment:
What is the plan for Split for later? Can you file a jira just for a place holder. Later we can update it with the approach.